frontend/src/metabase/questions/actions.ts

```diff
@@ -11,15 +11,15 @@ export interface LoadMetadataOptions {
 
 export const loadMetadataForCard =
   (card: Card, options?: LoadMetadataOptions) =>
-  (dispatch: Dispatch, getState: GetState) => {
+  async (dispatch: Dispatch, getState: GetState) => {
     const question = new Question(card, getMetadata(getState()));
-    const queries =
-      question.type() === "question"
-        ? [question.query()]
-        : [question.query(), question.composeDataset().query()];
```

Comment:
We cannot create an ad-hoc question before fetching metadata for the underlying question; doing so would fail

Reply:
This comment deserves to live in the code, imo.